Setup server with knex

leila100 opened the pull request.

leila100 committed:
Setup server with knex

leila100 committed:
Add the post route

endpoint: /api/zoo
Returns the id of the new zoo created
Throws an error if name is empty or doesn't exist

leila100 committed:
Add GET route

Endpoint: /api/zoos
Return list of zoos from database
Throws error if problem fetching zoos from database

leila100 committed:
Add GET route

Endpoint: /api/zoos/:id
Returns the zoo with the specific id
Throws an error if id is wrong or there's a problem with the database

leila100 committed:
Add DELETE route

Endpoint: /api/zoos/:id
Returns the number of zoos deleted
If it returns 0, the zoo with that id could not be found
Throws an error if there's a problem with the database

leila100 committed:
Add PUT route

Endpoint: /api/zoos/:id
Returns the number of zoos updated
If it returns 0, no zoos have been updated
Throws an error if problem with database

leila100 committed:
Refactor to separate zoos routes in their own folder

leila100 committed:
Add POST route to create a new bear

leila100 committed:
Add the rest of the CRUD routes for the bears resource

jperaltadev commented on bears/bears-router.js:
```diff
@@ -0,0 +1,99 @@
+router.get("/:id", (req, res) => {
+  db("bears")
+    .where({ id: Number(req.params.id) })
+    .first()
```
The `Number()` function isn't necessary as far as I'm aware

jperaltadev commented on zoos/zoos-router.js:
```diff
@@ -0,0 +1,99 @@
+
+router.post("/", (req, res) => {
+  const name = req.body.name
+  if (!name) {
```
You can also use a destructuring assignment if you want to use es6 syntax. `const { name } = req.body`

jperaltadev reviewed (approved):
Your endpoints look good! I would suggest modifying some of the status codes. `201` is for when you create a new resource so you should probably use `200` when updating a resource

leila100 committed:
Made changes based on feedback

Change status from 200 to 201 in PUT routes
Use destructuring to access the name value in req.body
